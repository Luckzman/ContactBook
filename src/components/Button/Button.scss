@import '../../styles/index.scss';

.btn{
  color: map-get($colors, white-color );
  background-color: map-get($colors, brand-color);
  padding: 10px;
  border: 1px solid transparent;
  border-radius: 5px;
  box-shadow: 1px 1px 5px 1px map-get($colors, grey-color );
  cursor: pointer;
  &:hover{
    box-shadow: none;
  }
}