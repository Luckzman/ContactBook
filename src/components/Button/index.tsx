import * as React from 'react';
import './Button.scss';

interface Props {
  children: React.ReactNode;
  className?: string;
  onClick?: (event: React.MouseEvent<HTMLElement>) => void;
  onSubmit?: (event: React.FormEvent<HTMLButtonElement>) => void;
  type?: 'button' | 'submit' | 'reset';
}
const Button: React.FC<Props> = ({ children, onClick, className, type = 'button', onSubmit }) => {
  return (
    <button className={`btn ${className}`} type={type} onClick={onClick} onSubmit={onSubmit}>
      {children}
    </button>
  );
};

export default Button;
