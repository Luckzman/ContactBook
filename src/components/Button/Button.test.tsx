import React from 'react';
import { shallow } from 'enzyme';
import Button from './';

it('renders without crashing', () => {
  const props = {
    onClick: jest.fn(),
  };
  const wrapper = shallow(<Button {...props}>Submit</Button>);
  expect(wrapper).toMatchSnapshot();
});
