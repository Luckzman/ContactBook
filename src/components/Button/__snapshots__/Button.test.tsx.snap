// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<button
  className="btn"
  onClick={[MockFunction]}
  type="button"
>
  Submit
</button>
`;
